pkg/proxy/topology.go

```diff
@@ -160,19 +164,23 @@ func canUseTopology(endpoints []Endpoint, svcInfo ServicePort, nodeLabels map[st
 		}
 	}
 
-	zone, ok := nodeLabels[v1.LabelTopologyZone]
-	if !ok || zone == "" {
-		klog.V(2).InfoS("Skipping topology aware endpoint filtering since node is missing label", "label", v1.LabelTopologyZone)
-		return false
-	}
-
+	zone, foundZone := nodeLabels[v1.LabelTopologyZone]
 	hasEndpointForZone := false
 	for _, endpoint := range endpoints {
 		if !endpoint.IsReady() {
 			continue
 		}
+
+		// If any of the endpoints do not have zone hints, we bail out
 		if endpoint.ZoneHints().Len() == 0 {
-			klog.V(2).InfoS("Skipping topology aware endpoint filtering since one or more endpoints is missing a zone hint", "endpoint", endpoint)
+			return false
```

Comment:
I think I'd leave this log in place for debugging but drop it down to something like 7. It is still useful to understand _why_ topology isn't working as expected, but I think it's fair to say that this will usually just be noise.

Reply:
agree. will keep it and switch to 7

Reply:
@gauravkghildiyal corrected me here, I think removing this log line is the best path forward for now. At some point in the future we could add a check for if hints are only set on a portion of endpoints for a Service, but that's out of scope for this PR. Removing is appropriate for now.